### PR TITLE
feat(linter): add the import/no_named_default rule

### PR DESCRIPTION
**Pull request by Spoutnik97**

### CODE REVIEW

File: crates/oxc_linter/src/rules/eslint/no_named_default.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            };
+            if import_name.name() == "default" {
+                ctx.diagnostic(no_named_default_diagnostic(entry.module_request.span()));
+            }
+        });
+    }
+}
+
+#[test]
+fn test() {
```
Comment (Sysix) [inline crates/oxc_linter/src/rules/eslint/no_named_default.rs:58]: for the reviewer, the original rule has differnt tests:

https://github.com/import-js/eslint-plugin-import/blob/main/tests/src/rules/no-named-default.js

**Comment by Sysix** [comment]
this is in the wrong folder? 

this should be: `crates/oxc_linter/src/rules/import/no_named_default.rs`

File: crates/oxc_linter/src/rules.rs [hunk 2]
```diff
@@ -533,6 +534,7 @@ oxc_macros::declare_all_lint_rules! {
     eslint::max_classes_per_file,
     eslint::max_lines,
     eslint::max_params,
+    eslint::no_named_default,
```
Comment (Boshen) [inline crates/oxc_linter/src/rules.rs:537]: This should be put under the `import` folder.

**Review by Boshen — CHANGES_REQUESTED** [review]

File: crates/oxc_linter/src/rules/eslint/no_named_default.rs (hunk 1 above)
Comment (Spoutnik97) [reply crates/oxc_linter/src/rules/eslint/no_named_default.rs]: I edited the exemple to use the one in the original test. 
I just omit the typeof test. I don't know how to make it pass without decreasing performances

**Review by camc314 — APPROVED** [review]
Thanks!

File: crates/oxc_linter/src/rules/eslint/no_named_default.rs (hunk 1 above)
Comment (camc314) [reply crates/oxc_linter/src/rules/eslint/no_named_default.rs]: can you include the typeof test and just comment it out with a reason? else it's confuring as to why it's commented out

**Review by Boshen — APPROVED** [review]

**Pull request merged**